src/main/java/io/stargate/sgv2/jsonapi/service/resolver/ValidateVectorize.java

```diff
@@ -0,0 +1,408 @@
+package io.stargate.sgv2.jsonapi.service.resolver;
+
+import io.stargate.sgv2.jsonapi.api.model.command.impl.VectorizeConfig;
+import io.stargate.sgv2.jsonapi.config.OperationsConfig;
+import io.stargate.sgv2.jsonapi.exception.ErrorCodeV1;
+import io.stargate.sgv2.jsonapi.exception.JsonApiException;
+import io.stargate.sgv2.jsonapi.service.embedding.configuration.EmbeddingProvidersConfig;
+import io.stargate.sgv2.jsonapi.service.embedding.configuration.ProviderConstants;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@ApplicationScoped
+public class ValidateVectorize {
```

Comment:
naming: if this is a validator object, shouldn't it be named something like "Vectorize[Config]Validator" (or "VectorizeServiceValidator")?